### PR TITLE
20kly: init at 1.4

### PR DESCRIPTION
**Pull request by fgaz**
###### Motivation for this change

The classic "20,000 Light-Years Into Space" game

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
  * +1.3M
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by mpickering** [comment]
@GrahamcOfBorg build 20kly

File: pkgs/games/20kly/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchurl, python2Packages }:
```
Comment (timokau) [inline pkgs/games/20kly/default.nix]: This should be `python` instead of `python2Packages`. The packages can then be accessed with `python.pkgs`. If it is not python3 compatible, you should add `disabled = !isPy3`.

Comment (timokau) [reply pkgs/games/20kly/default.nix]: Another one of my style preferences: I'd format the argument list like this:

```
{ stdenv
, fetchurl
, python2Packages
}:
```

Because of the same reasons as the list formatting in #45057.

File: pkgs/games/20kly/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  };
+
+  patchPhase = ''
+    substituteInPlace lightyears \
+      --replace \
+        "LIGHTYEARS_DIR = \".\"" \
+        "LIGHTYEARS_DIR = \"$out/share\""
+  '';
+
+  propagatedBuildInputs = with python2Packages; [ python pygame ];
```
Comment (timokau) [inline pkgs/games/20kly/default.nix]: Again, I don't think `python` belongs here.

File: pkgs/games/20kly/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  buildPhase = "python -O -m compileall .";
+
+  installPhase = ''
+    mkdir -p $out/share
```
Comment (timokau) [inline pkgs/games/20kly/default.nix]: Please quote shell variables.

File: pkgs/games/20kly/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    cp -r audio code data lightyears $out/share
+    install -Dm755 lightyears $out/bin/lightyears
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A steampunk-themed strategy game where you have to manage a steam supply network";
+    homepage = "http://jwhitham.org.uk/20kly/";
```
Comment (timokau) [inline pkgs/games/20kly/default.nix]: Quoting unnecessary

**Review by timokau — CHANGES_REQUESTED** [review]

**Comment by fgaz** [comment]
@timokau all done

@mpickering it's `_20kly`, since it starts with a number

**Review by timokau — APPROVED** [review]

File: pkgs/games/20kly/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{ stdenv
+, fetchurl
+, python }:
+
+python.pkgs.buildPythonApplication rec {
+  pname = "20kly";
+  version = "1.4";
+  format = "other";
+  disabled = !(python.isPy3 or true);
```
[the remark this answers is not in the document]
Comment (fgaz) [reply pkgs/games/20kly/default.nix]: On python2 it's null

**Comment by fgaz** [comment]
@timokau Nope, since it isn't a library (see the nixpkgs docs, python section)

File: pkgs/games/20kly/default.nix (hunk 5 above)
[the remark this answers is not in the document]
Comment (fgaz) [reply pkgs/games/20kly/default.nix]: `x or y` _should_ be sugar for `if x==null then y else x`, but I can't find where it's documented, if it's documented at all

[the remark this answers is not in the document]
Comment (timokau) [reply pkgs/games/20kly/default.nix]: > On python2 it's null

Oh? That doesn't seem right.

And what do you want to express? Shouldn't it be `disabled = !isPy2`, since you want to disable for everything that isn't python 2?

[the remark this answers is not in the document]
Comment (timokau) [reply pkgs/games/20kly/default.nix]: I don't see anybody else using `or null`. Does it make a difference if you add `isPy2` to the argument list instead of using `python.isPy2`?

[the remark this answers is not in the document]
Comment (fgaz) [reply pkgs/games/20kly/default.nix]: > I don't see anybody else using or null

```
git grep " or true"
git grep " or false"
```

> Shouldn't it be disabled = !isPy2, since you want to disable for everything that isn't python 2

Oops, you are right.. it should be `python.isPy3 or false` or `!(python.isPy2 or true)`... I wonder why it worked then

[the remark this answers is not in the document]
Comment (timokau) [reply pkgs/games/20kly/default.nix]: > git grep " or true"

Yes but not in that context (`git grep "disabled =`)

[the remark this answers is not in the document]
Comment (timokau) [reply pkgs/games/20kly/default.nix]: Did you try using `isPyX` from the argument list instead?

[the remark this answers is not in the document]
Comment (timokau) [reply pkgs/games/20kly/default.nix]: Well there is `python.pkgs.callPackage` (which is used in `python-packages.nix`). But outside of `python-packages` you'd have to choose which `python` to use. Which makes the point about `disabled` kind of moot.

TLDR: I'm probably just not qualified to review a python application, sorry for the noise. I hope @domenkozar will add his opinion.

**Comment by timokau** [comment]
I just noticed that I pinged the wrong person, I meant @dotlambda.
